src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,558 @@
+        {
+            const string text1 = @"
+class C
+{
+    internal virtual void M1() { }
+}
+partial class D : C
+{
+    internal sealed override partial void M1();
+    internal override partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (9,36): error CS9058: Both partial method declarations must be 'sealed' or neither may be 'sealed'.
+                //     internal override partial void M1() { }
+                Diagnostic(ErrorCode.ERR_PartialMethodSealedDifference, "M1").WithLocation(9, 36)
+            );
+        }
+
+        [Fact]
+        public void Extern_LangVersion()
+        {
+            const string text1 = @"
+using System.Runtime.InteropServices;
+
+partial class C
+{
+    static partial void M1();
+
+    [DllImportAttribute(""something.dll"")]
+    static extern partial void M1();
+}";
+            var comp = CreateCompilation(text1);
+            comp.VerifyDiagnostics(
+                // (9,32): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     static extern partial void M1();
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M1").WithArguments("extended partial methods").WithLocation(9, 32));
+
+            comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void New_LangVersion()
+        {
+            const string text1 = @"
+class C
+{
+    internal void M1() { }
+}
+
+partial class D : C
+{
+    new partial void M1();
+    new partial void M1() { }
+}";
+            var comp = CreateCompilation(text1);
+            comp.VerifyDiagnostics(
+                // (9,22): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     new partial void M1();
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M1").WithArguments("extended partial methods").WithLocation(9, 22),
+                // (10,22): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     new partial void M1() { }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M1").WithArguments("extended partial methods").WithLocation(10, 22));
+
+            comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void New_NoImpl()
+        {
+            const string text1 = @"
+class C
+{
+    internal void M1() { }
+}
+
+partial class D : C
+{
+    new partial void M1();
```

Comment:
I've been contemplating the behavior and precedent around 'new' and have been thinking that maybe we should just generally relax the requirements around 'new': adding 'new' doesn't require providing an implementation part, and both parts will behave as if 'new' was specified if either part has a 'new' keyword. Since the purpose of 'new' is just to conspicuously call out that we are not overriding something.

Just for context, in the shipped language there is no recourse if you want to declare a partial method with the same name as an accessible base member.

```cs
class C
{
    internal void M1() { }
}
partial class D : C
{
    // warning CS0108: 'D.M1()' hides inherited member 'C.M1()'. Use the new keyword if hiding was intended.
    partial void M1();
}
```